digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -708,18 +699,23 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
   def updateProjectLinkStatus(projectId: Long, linkIds: Set[Long], linkStatus: LinkStatus, userName: String): Boolean = {
     withDynTransaction{
       val projectLinks = withGeometry(ProjectDAO.getProjectLinks(projectId))
-      val (updatedProjectLinks, unchangedProjectLinks) = projectLinks.partition(pl => linkIds.contains(pl.linkId))
-      if (linkStatus == LinkStatus.Terminated)
-        ProjectDAO.updateProjectLinksToDB(updatedProjectLinks.map(_.copy(status=linkStatus, calibrationPoints = (None, None),
-          startAddrMValue = 0L, endAddrMValue = 0L)), userName)
+      val (updatedProjectLinks, unchangedProjectLinks) = projectLinks.filterNot(pl=> pl.status == LinkStatus.Terminated ).partition(pl => linkIds.contains(pl.linkId))
+      if (linkStatus == LinkStatus.Terminated) {
+        //Fetching road addresses in order to obtain the original addressMValues, since we may not have those values on project_link table, after previous recalculations
+        val roadAddresses = RoadAddressDAO.fetchByLinkId(updatedProjectLinks.map(pl => pl.linkId).toSet)
+        val updatedPL = updatedProjectLinks.map(pl => {
+          val roadAddress = roadAddresses.find(_.linkId == pl.linkId)
+          pl.copy(startAddrMValue = roadAddress.get.startAddrMValue, endAddrMValue = roadAddress.get.endAddrMValue)
```

Comment:
roadAddress could be empty here in anomalous cases here "val roadAddress = roadAddresses.find(_.linkId == pl.linkId)" is None. The "roadAddress.get" could give problems

Reply:
Termination and transfer has always a link with a link id that remains the same. Should there not be one then we have problems elsewhere in the code.

Also VIITE-697 will affect this and make sure that one road address row is found for a project link (unless LinkStatus == New)